packages/framework/attributor/src/test/attribution/documents/default/attributor-lz4-and-delta-snap.json

```diff
@@ -10,7 +10,7 @@
m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2},{\"pos1\":31,\"pos2\":32,\"props\":{\"prop2\":\"v1BgG\",\"prop1\":\"m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2},{\"pos1\":31,\"pos2\":33,\"props\":{\"prop2\":\"v1BgG\",\"prop1\":\"m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2},{\"pos1\":31,\"pos2\":32,\"props\":{\"prop2\":\"v1BgG\",\"prop1\":\"m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2},{\"pos1\":31,\"pos2\":32,\"props\":{\"prop2\":\"v1BgG\",\"prop1\":\"m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2},{\"pos1\":31,\"pos2\":33,\"props\":{\"prop2\":\"v1BgG\",\"prop1\":\"m4xRB\",\"prop3\":\"8e12Q\"},\"type\":2}],\"type\":3},\"referenceSequenceNumber\":79,\"type\":\"op\",\"sequenceNumber\":80,\"minimumSequenceNumber\":76},{\"clientId\":\"a2df62bb-461f-4c2e-a3ad-fc5df463c464\",\"clientSequenceNumber\":28,\"contents\":{\"pos1\":31,\"pos2\":36,\"type\":1},\"referenceSequenceNumber\":80,\"type\":\"op\",\"sequenceNumber\":81,\"minimumSequenceNumber\":76},{\"clientId\":\"d3fdea22-7906-496f-9fc5-0bf988954452\",\"clientSequenceNumber\":28,\"contents\":{\"ops\":[{\"pos1\":30,\"pos2\":40,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":35,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":33,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":34,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":34,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":32,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":34,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":33,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":38,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":40,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2}],\"type\":3},\"referenceSequenceNumber\":81,\"type\":\"op\",\"sequenceNumber\":82,\"minimumSequenceNumber\":76},{\"clientId\":\"d3fdea22-7906-496f-9fc5-0bf988954452\",\"clientSequenceNumber\":29,\"contents\":{\"pos1\":31,\"seg\":\"yc5Bz\",\"type\":0},\"referenceSequenceNumber\":82,\"type\":\"op\",\"sequenceNumber\":83,\"minimumSequenceNumber\":76},{\"clientId\":\"d3fdea22-7906-496f-9fc5-0bf988954452\",\"clientSequenceNumber\":30,\"contents\":{\"pos1\":36,\"seg\":\"mQny\",\"type\":0},\"referenceSequenceNumber\":83,\"type\":\"op\",\"sequenceNumber\":84,\"minimumSequenceNumber\":76},{\"clientId\":\"25adaa92-9220-49e2-a852-3c7e97ac9d1c\",\"clientSequenceNumber\":24,\"contents\":{\"ops\":[],\"type\":3},\"referenceSequenceNumber\":84,\"type\":\"op\",\"sequenceNumber\":85,\"minimumSequenceNumber\":76},{\"clientId\":\"a2df62bb-461f-4c2e-a3ad-fc5df463c464\",\"clientSequenceNumber\":29,\"contents\":{\"pos1\":4,\"seg\":\"x2NAdmR\",\"type\":0},\"referenceSequenceNumber\":85,\"type\":\"op\",\"sequenceNumber\":86,\"minimumSequenceNumber\":76,\"timestamp\":1665600040000}]"
```

Comment:
Is this related to this PR? All I see when skimming the diff is a timestamp.

Reply:
It appears caused by this PR, but as you've mentioned is largely harmless and just adds the timestamp field to the snapshot. It's not clear to me why this occurs.

Reply:
Feels like the kind of thing that's worth understanding why. As we both know, there's relevant code here that roundtrips through JSON string (or not), maybe the PR changes the exact logic there in a way that triggers serialization to change somehow.

cc @Abe27342 in case he has comments on the purpose of these snapshots and why we might care about an apparently harmless change.

Reply:
It's not really concerning to me. Real SharedString snapshots will have timestamps since they just stick the `ISequencedDocumentMessage` in their format (which is also why we have some back-compat code dealing with "term"). Our mocks--which this test suite use--don't set up timestamps by default. Since attribution cares about timestamps, it sets them up on its own in a bit of a kludgy way, see below:

![image](https://github.com/microsoft/FluidFramework/assets/5222607/9a5d4aa5-b6e6-44fe-8e0c-d06d4dc01099)

This kludge ended up not carrying over into the catchup ops part of the snapshot before, probably due to some object copies that you've now removed, and it's plausible that they do now. This test suite is pretty much only concerned with attribution snapshot format anyway, which you haven't made changes to.